### PR TITLE
feat: tidy up tests and make datasources compatible with kubernetes deployment

### PR DESCRIPTION
**Pull request by raymondfeng**
Spin-off from https://github.com/strongloop/loopback4-example-shopping/pull/268

1. Improve tests to avoid side-efforts (make sure setup code to execute in mocha hooks)
2. Make sure datasources are disconnected upon `app.stop` 
3. Add logic to `*.datasources.ts` to accept configuration from env vars by kubernetes

### CODE REVIEW

File: packages/shopping/src/__tests__/acceptance/authentication.acceptance.ts [hunk 1]
```diff
@@ -196,7 +200,8 @@ describe('authentication services', () => {
   });
 
   async function setupApp() {
-    app = await setupApplication();
+    const appWithClient = await setupApplication();
```
Comment (emonddr) [inline packages/shopping/src/__tests__/acceptance/authentication.acceptance.ts]: @raymondfeng , how were these tests working before if the wrong object was being assigned to `app`?

File: bin/rebuild-package-locks.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env node
```
Comment (emonddr) [inline bin/rebuild-package-locks.js]: @raymondfeng , why do we need to rebuild package locks?
Is this a script we should make available to anyone building an LB4 app?

File: packages/shopping/src/datasources/mongo.datasource.ts [hunk 3]
```diff
@@ -3,17 +3,34 @@
...
 
+function updateConfig(dsConfig: AnyObject) {
+  if (process.env.KUBERNETES_SERVICE_HOST) {
+    dsConfig.host = process.env.SHOPPING_APP_MONGODB_SERVICE_HOST;
+    dsConfig.port = +process.env.SHOPPING_APP_MONGODB_SERVICE_PORT!;
+  }
+  return dsConfig;
+}
+
+@lifeCycleObserver('datasource')
```
Comment (emonddr) [inline packages/shopping/src/datasources/mongo.datasource.ts]: Very cool use of life cycle observers.

File: packages/shopping/src/__tests__/acceptance/authentication.acceptance.ts (hunk 1 above)
Comment (raymondfeng) [reply packages/shopping/src/__tests__/acceptance/authentication.acceptance.ts]: Haha, the test was moved from `unit` to `acceptance` and it used to have a different `helper`.

File: packages/shopping/src/datasources/mongo.datasource.ts (hunk 3 above)
Comment (raymondfeng) [reply packages/shopping/src/datasources/mongo.datasource.ts]: We now generate datasources in this pattern with `lb4 datasource`.

File: bin/rebuild-package-locks.js (hunk 2 above)
Comment (raymondfeng) [reply bin/rebuild-package-locks.js]: The shopping project is set up as a lerna monorepo and it requires a few steps to rebuild package-lock files.

**Pull request merged**